contracts/contracts/core/vault/src/main.sw

```diff
@@ -998,72 +617,64 @@ impl Vault for Contract {
     #[storage(read, write)]
     fn increase_position(
```

Comment:
It will be hard to implement slippage protection on that.
We should return execution price in this call, so trader can revert if it doesn't satisfy his desired fill price.
Same for `decrease_position`.